stl/inc/xmemory

```diff
@@ -804,6 +804,7 @@ public:
     }
 };
 
+#if !_HAS_CXX20

```

Comment:
The `allocator<void>` specialization was removed from the standard only because the primary template was able to do all the same things. Given that we have escape hatches, I think this needs to always stay.

Reply:
I cannot follow. If it has no purpose why should it stay?

Reply:
It has a purpose: if the user has used one of the escape hatches to put the typedefs back on the primary template, then it would make `std::allocator<void>` ill formed. The only reason for this specialization to exist is to make naming that type well formed; if we always leave this here, that will always be well formed. Even if the primary template could be used, less conditional compilation is better.

Reply:
Kept it for good